### PR TITLE
'Implement user view of game library'

### PR DESCRIPTION
**Pull request by JannisAziz**
Frontend:
- [x] + axios request getUser via new UserService.ts
- [x] + display user.savedGames via new SavedGamesList.tsx
- [x] + data models 'SavedGame' & 'User'

Backend:
- [x] + request mapping for getUser via new UserController.java
- [x] + controller<->repository function via new UserService.java
- [x] + mongo user repository via new UserRepository.java
- [x] + data models 'SavedGame' & 'User'

Config:
- [x] + frontend proxy (package.json)
- [x] + mongo uri & server port (application.yml)
- [x] + dependency: 'spring data' for mongoDB

### CODE REVIEW

File: backend/src/main/java/de/jannisaziz/backend/controllers/UserController.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import de.jannisaziz.backend.services.UserService;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+@RestController
+@RequestMapping("/users")
```
Comment (Flooooooooooorian) [inline backend/src/main/java/de/jannisaziz/backend/controllers/UserController.java:13]:
```suggestion
@RequestMapping("/api/users")
```

File: backend/src/main/java/de/jannisaziz/backend/models/SavedGame.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.data.annotation.Id;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class SavedGame {
```
Comment (Flooooooooooorian) [inline backend/src/main/java/de/jannisaziz/backend/models/SavedGame.java:11]: SavedGame? Do notSavedGames exist?

File: frontend/src/components/SavedGamesList.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+export default function SavedGamesList() {
+
+    const [games, setGames] = useState<SavedGame[]>([])
+
+    useEffect(() => {
+        getUserById("TEST_ID")
+            .then((user: User) => setGames(user.savedGames))
+            .catch(console.error)
+    }, [])
```
Comment (Flooooooooooorian) [inline frontend/src/components/SavedGamesList.tsx:15]: If you only need the games from a specific/loggedin user you maybe want to create an enpoint for that?

**Review by Flooooooooooorian — APPROVED** [review]

File: backend/src/main/java/de/jannisaziz/backend/models/SavedGame.java (hunk 2 above)
Comment (JannisAziz) [reply backend/src/main/java/de/jannisaziz/backend/models/SavedGame.java:11]: This is a game that the user adds/saves to their library

File: backend/src/main/java/de/jannisaziz/backend/controllers/UserController.java (hunk 1 above)
Comment (JannisAziz) [reply backend/src/main/java/de/jannisaziz/backend/controllers/UserController.java:13]: Thank you, will change!

File: frontend/src/components/SavedGamesList.tsx (hunk 3 above)
Comment (JannisAziz) [reply frontend/src/components/SavedGamesList.tsx:15]: I will consider that 👍

**Pull request merged**